structuralcodes/geometry/_geometry.py

```diff
@@ -93,6 +104,7 @@ def __init__(
         point: t.Union[Point, ArrayLike],
         diameter: float,
         material: t.Union[Material, ConstitutiveLaw],
+        density: t.Optional[float] = None,
```

Comment:
Should we perhaps specify that the value of density is only used if material is `ConstitutiveLaw`?